tools/install/DynamoInstaller.iss

```diff
@@ -187,66 +185,36 @@ begin
   ExtractTemporaryFile('RevitAddinUtility.dll');
   ExtractTemporaryFile('DynamoAddinGenerator.exe');
 
+  result := true
   // check if there is a valid revit installation on this machine, if not - fail
-  if (RevitInstallationExists('Revit2016') or RevitInstallationExists('Revit2015') or RevitInstallationExists('Revit2014') or RevitInstallationExists('VasariBeta3')) then
+  if not (RevitInstallationExists('Revit2016') or RevitInstallationExists('Revit2015') or RevitInstallationExists('Revit2014')) then
     begin
-    result := true;
+	  MsgBox('Dynamo requires an installation of Revit 2014, Revit 2015 or Revit 2016 in order to proceed!', mbCriticalError, MB_OK);
+      result := false;
     end
-  else
-    begin
-    MsgBox('Dynamo requires an installation of Revit 2014, Revit 2015, Revit 2016, or Vasari Beta 3 in order to proceed!', mbCriticalError, MB_OK);
-    result := false;
-    end;
-end;
-
-/////////////////////////////////////////////////////////////////////
-function IsEXEInstalled(): Boolean;
-begin
-  Result := (GetUninstallStringForEXE() <> '');
-end;
-
-/////////////////////////////////////////////////////////////////////
-function UnInstallOldEXE(sUnInstallString: String): Integer;
-var
-  iResultCode: Integer;
-  installStr: String;
-begin
-// Return Values:
-// 1 - uninstall string is empty
-// 2 - error executing the UnInstallString
-// 3 - successfully executed the UnInstallString
 
-  // default return value
-  Result := 0;
-
-  if sUnInstallString <> '' then begin
-    sUnInstallString := RemoveQuotes(sUnInstallString);
-    installStr := '/SILENT /NORESTART /SUPPRESSMSGBOXES ';
-
-    if Exec(sUnInstallString, installStr + updateFlag,'', SW_HIDE, ewWaitUntilTerminated, iResultCode) then
-      Result := 3
-    else
-      Result := 2;
-  end else
-    Result := 1;
-end;
-
-function UnInstallOldMSI(sUnInstallString: String): Integer;
-var
-  iResultCode: Integer;
-begin
-  // default return value
-  Result := 0;
-
-  if sUnInstallString <> '' then begin
-    sUnInstallString := RemoveQuotes(sUnInstallString);
-	
-	if Exec('MsiExec.exe', sUnInstallString + ' /quiet','', SW_SHOWNORMAL, ewWaitUntilTerminated, iResultCode) then
-      Result := 3
-    else
-      Result := 2;
-  end else
-    Result := 1;
+  sUnInstPath := ExpandConstant('Software\Microsoft\Windows\CurrentVersion\Uninstall\{#ProductName} {#Major}.{#Minor}');
+  sUninstallString := '';
+  RegQueryStringValue(HKLM, sUnInstPath, 'UnInstallString', sUninstallString);
+    if (sUninstallString <> '') then
+	begin
+		if not RegQueryDWordValue(HKLM, sUnInstPath, 'RevVersion', revision) then
+			begin
+				MsgBox('The installed Dynamo version is corrupted. Aborting...', mbInformation, MB_OK);
```

Comment:
Couldn't determine the revision number for already installed {#ProductName} {#Major}.{#Minor} In order to proceed with the installation, you need to uninstall {#ProductName} {#Major}.{#Minor} manually.